src/statusEditMode.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as vscode from "vscode";
+import { HexDocumentEditOp } from "../shared/hexDocumentModel";
+import { Disposable, DisposableValue } from "./dispose";
+import { HexDocument } from "./hexDocument";
+import { HexEditorRegistry } from "./hexEditorRegistry";
+
+const numberFormat = new Intl.NumberFormat();
+
+/**
+ * this is a class to represent the status bar item that displays the edit mode
+ *  - Replace or Insert
+ *
+ * @class StatusSelectionCount
+ */
+export default class StatusEditMode extends Disposable {
+	private readonly item: vscode.StatusBarItem;
+	private readonly docChangeListener = this._register(new DisposableValue());
+
+	constructor(registry: HexEditorRegistry) {
+		super();
+
+		this.item = this._register(
+			vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Right, 99),
+		);
+		this.item.tooltip = vscode.l10n.t("Select Edit Mode");
+		this.item.command = "hexEditor.selectEditMode";
```

Comment:
Since there's only two modes, I think clicking on them should immediately change the mode instead of opening a quickpick.

You can register a separate internal command for this (just call registerCommand without adding it in the package.json)

Reply:
I changed the quickpick, but I didn't add the internal command since I think it would be better just to have a single command to change the edit mode. But I do not mind adding back the select command.